extension/wrapper.cpp

```diff
@@ -144,66 +183,130 @@ Satrec__sgp4(PyObject *self, PyObject *args)
 static PyMethodDef Satrec_methods[] = {
     {"twoline2rv", (PyCFunction)Satrec_twoline2rv, METH_VARARGS | METH_CLASS,
      PyDoc_STR("Initialize the record from two lines of TLE text.")},
+    {"sgp4init", (PyCFunction)Satrec_sgp4init, METH_VARARGS,
+     PyDoc_STR("Initialize the record from orbital elements.")},
     {"sgp4", (PyCFunction)Satrec_sgp4, METH_VARARGS,
-     PyDoc_STR("Given minutes since epoch, return position and velocity.")},
+     PyDoc_STR("Given a modified julian date, return position and velocity.")},
     {"_sgp4", (PyCFunction)Satrec__sgp4, METH_VARARGS,
-     PyDoc_STR("Given an array of minutes since epoch, return position and velocity arrays.")},
+     PyDoc_STR("Given an array of modified julian dates, return position and velocity arrays.")},
+    {"sgp4_tince", (PyCFunction)Satrec_sgp4, METH_VARARGS,
+     PyDoc_STR("Given minutes since epoch, return position and velocity.")},
     {NULL, NULL}
 };
 
 #define O(member) offsetof(SatrecObject, satrec.member)
 
 static PyMemberDef Satrec_members[] = {
-    /* Listed in the order they appear in a TLE record. */
+    /* Listed in the order they appear in a TLE record. 
+    Writeable so that sgp4init() may initialize from an empty Satrec() call */
```

Comment:
`AttributeError: readonly attribute`

Reply:
Perhaps a poor example.  Take a case were you're updating the elements to adjust an epoch.

Reply:
@brandon-rhodes I've found an acceptable implementation of `sgp4init()` which leaves the variables as read-only, exposes additional (read-only) variables for convenience with **one key exception** - the epoch-related variables:
1. satrec.jdsatepoch
1. satrec.jdsatepochF
1. satrec.epochyr
1. satrec.epochdays

When a `Satrec()` is initialized with twoline2rv, these variables are set/derived from the epoch data in the TLE - giving a convenient internal reference for what epoch the `Satrec` is associated with. 

Vallado's `sgp4init()` offers no equivalent, and no internal functions with which to call in order to derive the variables - the SGP4epoch -> MJD epoch is easy enough, but converting either of those to a year and DOY are not possible, as there is not an inverse of `days2mdhms()`

Since we're treating the Vallado C-code as "read-only", I would propose allowing the above 4 variables to be writable such that `sgp4init()` users can create a workflow and structure equivalent to `twoline2rv()` calls (if they care to).  This would appear to be safe, as these variables are only set and referenced by `twoline2rv()`, but not used by any other SGP4 function.

`twoline2rv()` would automatically overwrite anything with variables consistent with the provided TLE, and anyone who is using them via `sgp4init()` would have had to take the explicit action to set them as well.

Reply:
Resolved this by having model.Satrec.sgp4init() and wrapper.c:sgp4init() set satrec.jdsatepoch and satrec.jdsatepochF so it will work the same everywhere in the V2 code.

Let epochyr and epochdays unset (even though twoline2rv sets them), as there's no existing code within the code base to get epochdays, and it's not used by any of the other functions.